### PR TITLE
SW-1223 don't show the update failed info message on boot

### PR DESCRIPTION
**Pull request by JFSolution**
add user clicked update flag
redirect softwareupdateplugin check for update to our endpoint if the user clicked on it
refactor fallback update config for offline

### CODE REVIEW

File: octoprint_mrbeam/__init__.py [hunk 1]
```diff
@@ -738,6 +739,15 @@ def calibration_tool_mode(self):
         self._fixEmptyUserManager()
         return ret
 
+    @property
+    def user_checks_for_updates(self):
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/__init__.py]: suggest the following refactor:
`user_checks_for_updates` --> `explicit_Update_Check`

File: octoprint_mrbeam/__init__.py [hunk 2]
```diff
@@ -738,6 +739,15 @@ def calibration_tool_mode(self):
...
+        return self._user_checks_for_updates
+
+    @user_checks_for_updates.setter
+    def user_checks_for_updates(self, value):
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/__init__.py]: Since it is a boolean value, I would suggest to create two setters, one for set and one for clear

File: octoprint_mrbeam/rest_handler/update_handler.py [hunk 3]
```diff
@@ -12,5 +12,12 @@ class UpdateRestHandlerMixin:
 
     @octoprint.plugin.BlueprintPlugin.route("/info/update", methods=["POST"])
     def update_update_informations(self):
-        reload_update_info(self)
-        return NO_CONTENT
+        data = None
+        clicked_by_user = False
+        if hasattr(request, "json") and request.json:
+            data = request.json
+            clicked_by_user = data.get("user", False)
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/rest_handler/update_handler.py:18]: Are you sure you want to default to False ?

File: octoprint_mrbeam/software_update_information.py [hunk 4]
```diff
@@ -54,6 +54,30 @@ class SWUpdateTier(Enum):
 BEAMOS_LEGACY_VERSION = "0.14.0"
 BEAMOS_LEGACY_DATE = date(2018, 1, 12)  # still used in the migrations
 
+FALLBACK_UPDATE_CONFIG = {
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/software_update_information.py:57]: Is this change related to the fix ?

File: octoprint_mrbeam/software_update_information.py [hunk 5]
```diff
@@ -168,32 +193,7 @@ def get_update_information(plugin):
...
-                "findmymrbeam": {
-                    "name": "OctoPrint-FindMyMrBeam",
-                    "type": "github_commit",
-                    "user": "",
-                    "repo": "",
-                    "pip": "",
-                },
-            },
-        },
+        {},
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/software_update_information.py:196]: I don't understand what happened here

File: octoprint_mrbeam/software_update_information.py [hunk 6]
```diff
@@ -229,14 +229,15 @@ def switch_software_channel(plugin, channel):
     Returns:
         None
     """
+    _logger.debug("switch_software_channel")
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/software_update_information.py:232]: Could you please add at the begging of the log something  like "Software update information: "
so that it can be clear in the logs?

File: octoprint_mrbeam/software_update_information.py [hunk 7]
```diff
@@ -229,14 +229,15 @@ def switch_software_channel(plugin, channel):
...
     old_channel = plugin._settings.get(["dev", "software_tier"])
     if channel in software_channels_available(plugin) and channel != old_channel:
         _logger.info("Switching software channel to: {channel}".format(channel=channel))
         plugin._settings.set(["dev", "software_tier"], channel)
         reload_update_info(plugin)
 
 
-def reload_update_info(plugin):
+def reload_update_info(plugin, clicked_by_user=False):
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/software_update_information.py:240]: Not sure why we have `clicked_by_user` and _user_checks_for_updates why not use one of them ?

**Review by ahmed-mrbeam — COMMENTED** [review]
Partially reviewed

File: octoprint_mrbeam/rest_handler/update_handler.py (hunk 3 above)
Comment (JFSolution) [reply octoprint_mrbeam/rest_handler/update_handler.py:18]: Yes, if it is not set the default should be false, as we only set it to true if the users clicks on it and otherwise leave this data element empty

File: octoprint_mrbeam/software_update_information.py (hunk 6 above)
Comment (JFSolution) [reply octoprint_mrbeam/software_update_information.py:232]: this will be added from the logger it selfe as it logs everything from this file as `octoprint.plugins.mrbeam.software_update_information`

File: octoprint_mrbeam/software_update_information.py (hunk 4 above)
Comment (JFSolution) [reply octoprint_mrbeam/software_update_information.py:57]: Yes I needed to refactor this so I am still able to give back the fallback update config

File: octoprint_mrbeam/software_update_information.py (hunk 5 above)
Comment (JFSolution) [reply octoprint_mrbeam/software_update_information.py:196]: this is releated to the refactoring https://github.com/mrbeam/MrBeamPlugin/pull/1463/files#r859892304
I refactored it so the _set_info_from_cloud_config method will return the `FALLBACK_UPDATE_CONFIG` if something is not correct instead of returning it only if there was a connection problem

File: octoprint_mrbeam/software_update_information.py (hunk 7 above)
Comment (JFSolution) [reply octoprint_mrbeam/software_update_information.py:240]: clicked by user is only the input parameter of reload_update_info, that will then be saved to the user_checks_for_updates

File: octoprint_mrbeam/__init__.py (hunk 2 above)
Comment (JFSolution) [reply octoprint_mrbeam/__init__.py]: ok i can change that

File: octoprint_mrbeam/software_update_information.py [hunk 8]
```diff
@@ -245,6 +246,9 @@ def reload_update_info(plugin):
     Returns:
         None
     """
+    if clicked_by_user:
+        plugin.user_checks_for_updates = True
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/software_update_information.py]: please use the setter

File: octoprint_mrbeam/software_update_information.py [hunk 9]
```diff
@@ -320,17 +325,20 @@ def _set_info_from_cloud_config(plugin, tier, beamos_version, cloud_config):
                 f.write(json.dumps(sw_update_config))
         except (IOError, TypeError):
             plugin._logger.error("can't create update info file")
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/software_update_information.py]: Please add here that Update fallback config will be used

File: octoprint_mrbeam/static/js/software_channel_selector.js [hunk 10]
```diff
@@ -95,11 +95,12 @@ $(function () {
         // get the hook when softwareUpdate perform the Updatecheck to force the update on the normal button
         self.performCheck_copy = self.softwareUpdate.performCheck;
         self.softwareUpdate.performCheck= function(showIfNothingNew, force, ignoreSeen) {
-            self.reload_update_info();
             if (force !== undefined) {
                 force = true; //only forces the update check if it was disabled ("check for update" button press)
+                self.reload_update_info(showIfNothingNew, force, ignoreSeen, true); //if use clicked update forward to our reload endpoint
```
Comment (ahmed-mrbeam) [inline octoprint_mrbeam/static/js/software_channel_selector.js:100]: `showIfNothingNew` `ignoreSeen`   why using different naming convention here ?

File: tests/softwareupdate/test_cloud_config.py [hunk 11]
```diff
@@ -461,10 +438,11 @@ def test_cloud_confg_fileerror(
                 },
             )
             plugin = self.plugin
+            plugin.user_checks_for_updates = True
```
Comment (ahmed-mrbeam) [inline tests/softwareupdate/test_cloud_config.py]: Please use the setter

File: octoprint_mrbeam/__init__.py (hunk 1 above)
Comment (JFSolution) [reply octoprint_mrbeam/__init__.py]: I can do this

File: octoprint_mrbeam/__init__.py (hunk 2 above)
Comment (JFSolution) [reply octoprint_mrbeam/__init__.py]: will change it to set_explicit_update_check and clear_explicit_update_check

**Review by ahmed-mrbeam — APPROVED** [review]

**Review by ahmed-mrbeam — APPROVED** [review]
